marklogic/src/main/scala/quasar/physical/marklogic/fs/xdmitem.scala

```diff
@@ -21,52 +21,102 @@ import quasar.Data
+    case item: JSArray                  => jsonToData[F](item.asString)
+    case item: JSObject                 => jsonToData[F](item.asString)
+    case item: JsonItem                 => jsonToData[F](item.asString)
 
     case item: XdmAttribute             =>
       val attr = item.asW3cAttr
-      Data.Obj(ListMap(attr.getName -> Data.Str(attr.getValue)))
+      Data.singletonObj(attr.getName, Data.Str(attr.getValue)).point[F]
 
     // TODO: Inefficient for large data as it must be buffered into memory
-    case item: XdmBinary                => Data.Binary(ImmutableArray.fromArray(item.asBinaryData))
-    case item: XdmComment               => Data.NA
-    case item: XdmDocument              => SecureXML.loadString(item.asString) map (xml.toData) getOrElse Data.NA
-    case item: XdmElement               => SecureXML.loadString(item.asString) map (xml.toData) getOrElse Data.NA
-    case item: XdmProcessingInstruction => Data.NA
-    case item: XdmText                  => Data.Str(item.asString)
-    case item: XSAnyURI                 => Data.Str(item.asString)
-    case item: XSBase64Binary           => Data.Binary(ImmutableArray.fromArray(item.asBinaryData))
-    case item: XSBoolean                => Data.Bool(item.asPrimitiveBoolean)
-    case item: XSDate                   => ???
-    case item: XSDateTime               => ???
-    case item: XSDecimal                => Data.Dec(item.asBigDecimal)
-    case item: XSDouble                 => Data.Dec(item.asBigDecimal)
-    case item: XSDuration               => ???
-    case item: XSFloat                  => Data.Dec(item.asBigDecimal)
-    case item: XSGDay                   => ???
-    case item: XSGMonth                 => ???
-    case item: XSGMonthDay              => ???
-    case item: XSGYear                  => ???
-    case item: XSGYearMonth             => ???
-    case item: XSHexBinary              => Data.Binary(ImmutableArray.fromArray(item.asBinaryData))
-    case item: XSInteger                => Data.Int(item.asBigInteger)
-    case item: XSQName                  => Data.Str(item.asString)
-    case item: XSString                 => Data.Str(item.asString)
-    case item: XSTime                   => ???
-    case item: XSUntypedAtomic          => ???
-    case _                              => Data.NA // This case should not be hit although we can't prove it.
+    case item: XdmBinary                => bytesToData[F](item.asBinaryData)
+    case item: XdmComment               => Data.singletonObj("xdm:comment"  , Data.Str(item.asString)).point[F]
+    case item: XdmDocument              => xmlToData[F](item.asString)
+    case item: XdmElement               => xmlToData[F](item.asString)
+    case item: XdmProcessingInstruction => Data.singletonObj("xdm:processingInstruction", Data.Str(item.asString)).point[F]
+    case item: XdmText                  => Data._str(item.asString).point[F]
```

Comment:
Maybe it would be better to either use `Data.Str` everywhere or use `Data._str` everywhere.